### PR TITLE
Posix compat

### PR DESCRIPTION
**Pull request by ShellyHerself**
Should be enough compat to make tag loading work for libraries that depend on just supyr.

### CODE REVIEW

**Comment by MosesofEgypt** [comment]
Need to convert filepath to a string when passing to backup_and_rename_temp, otherwise it'll never get renamed.
https://github.com/MosesofEgypt/supyr_struct/blob/83cbaad2a6c0c391fecaf93918c0ecee9c65c9ab/tag.py#L545

**Comment by ShellyHerself** [comment]
> Need to convert filepath to a string when passing to backup_and_rename_temp, otherwise it'll never get renamed.
> https://github.com/MosesofEgypt/supyr_struct/blob/83cbaad2a6c0c391fecaf93918c0ecee9c65c9ab/tag.py#L545

Again, not true. Here is a test script that takes methods from all the modules used in `backup_and_rename_temp`
```py
import pathlib
import os

some_path = pathlib.Path("./test_file.txt")
some_new_path = pathlib.Path("./test_file2.txt")

os.rename(some_path, some_new_path)

with open(some_new_path, 'r') as f:
    print(str(some_new_path) + ':\n', *f.readlines(), sep='')

print("\n")

os.rename(some_new_path, some_path)

with open(some_path, 'r') as f:
    print(str(some_path) + ':\n', *f.readlines(), sep='')

print("\n")

print("using some_path in a format string: %s" % (some_path))
```
Output:
```
michelle@nessy:~/Tinkering$ python3 pathlib_test.py
test_file2.txt:
This is some text in test_file.txt

See, it works.


test_file.txt:
This is some text in test_file.txt

See, it works.


using some_path in a format string: test_file.txt
```

I have manually tested all this functionality to work. And this is also why we're getting unit tests soon. Because there is too many assumptions going on here about how versatile the standard libraries are. They're a lot smarter than you think.

**Comment by ShellyHerself** [comment]
There seems to be no issues here, this is ready for a merge.

**Comment by ShellyHerself** [comment]
After some research it turns out this is an issue with Python versions below 3.6. I have rewritten backup_and_rename_temp to function properly with path objects making sure it is compatible with 3.5.

And it now has a docstring!

**Comment by MosesofEgypt** [comment]
It is not a good idea to have both an object named "path" and one named "Path" in this namespace. Remove this line and change references to "path" to instead use "os.path"
https://github.com/MosesofEgypt/supyr_struct/blob/cb35aabd5aa142231c2d6a8dce1af1945fc3d2ce/util.py#L4

There are some functions that have "path" as their argument name, so make sure to ignore those.

**Review by MosesofEgypt — CHANGES_REQUESTED** [review]
It is not a good idea to have both an object named "path" and one named "Path" in this namespace. Remove this line and change references to "path" to instead use "os.path"
https://github.com/MosesofEgypt/supyr_struct/blob/cb35aabd5aa142231c2d6a8dce1af1945fc3d2ce/util.py#L2-L4
There are some functions that have "path" as their argument name, so make sure to ignore those.

This call to mkdir needs to provide `parents=True` or else the backup directory will not be created if nested in non-existent directories.
https://github.com/MosesofEgypt/supyr_struct/blob/cb35aabd5aa142231c2d6a8dce1af1945fc3d2ce/util.py#L60-L64

A complete exception catch in these exception blocks is not a good idea, as they will also catch keyboard interrupt and system exit exceptions. Please change it to `except Exception:`
https://github.com/MosesofEgypt/supyr_struct/blob/cb35aabd5aa142231c2d6a8dce1af1945fc3d2ce/util.py#L68-L75

File: util.py [hunk 1]
```diff
@@ -1,4 +1,7 @@
+import pathlib
+from pathlib import Path, PurePath, PurePosixPath, PureWindowsPath
 import os
+from os import path
```
Comment (ShellyHerself) [inline util.py]: It is not a good idea to have both an object named "path" and one named "Path" in this namespace. Remove this line and change references to "path" to instead use "os.path"

File: util.py [hunk 2]
```diff
@@ -32,47 +35,43 @@ def int_to_fourcc(value, byteorder='big', signed=False):
...
-        os.remove(filepath)
+    backuppath = Path(backuppath)
+
+    assert not backuppath.is_dir(), "backuppath cannot already exist as a directory."
+
+    if remove_old_backup and backuppath.exists():
+        backuppath.unlink()
+
+    if filepath.exists() and not backuppath.exists():
+        backuppath.parent.mkdir(exist_ok=True)
```
Comment (ShellyHerself) [inline util.py]: This call to mkdir needs to provide `parents=True` or else the backup directory will not be created if nested in non-existent directories.

File: util.py [hunk 3]
```diff
@@ -32,47 +35,43 @@ def int_to_fourcc(value, byteorder='big', signed=False):
...
-        os.rename(temppath, filepath)
+        temppath.rename(filepath)
         return
-    except Exception:
+    except:
         try:
-            os.rename(backuppath, filepath)
-        except Exception:
+            backuppath.rename(filepath)
+        except:
```
Comment (ShellyHerself) [inline util.py]: A complete exception catch in these exception blocks is not a good idea, as they will also catch keyboard interrupt and system exit exceptions. Please change it to `except Exception:`

**Review by MosesofEgypt — APPROVED** [review]
Everything looks good now. Lovely work.

**Review by MosesofEgypt — CHANGES_REQUESTED** [review]
is_in_dir fails to detect saving outside tags directory.
https://github.com/MosesofEgypt/supyr_struct/blob/8568a36f458d712fc0e6dfd7a76c72b79bfa8094/util.py#L138

Allows me to save this tag in the same directory as the tags dir, as well as one directory higher. I have not tested, but it should be assumed saving to higher directories is allowed by this behavior.
https://cdn.discordapp.com/attachments/329353207463673857/650090875438891028/unknown.png
https://cdn.discordapp.com/attachments/329353207463673857/650091109816467479/unknown.png

**Pull request merged**